packages/shared/src/components/buttons/ButtonV2.tsx

```diff
@@ -0,0 +1,193 @@
+import React, {
+  forwardRef,
+  HTMLAttributes,
+  ReactElement,
+  ReactNode,
+  Ref,
+} from 'react';
+import classNames from 'classnames';
+import { IconProps, IconSize } from '../Icon';
+import { Loader } from '../Loader';
+import { combinedClicks } from '../../lib/click';
+
+import { ColorName as ButtonColor } from '../../styles/colors';
+
+export enum ButtonSize {
+  XLarge = 'xlarge', // only used for iconOnly buttons
+  Large = 'large',
+  Medium = 'medium',
+  Small = 'small',
+  XSmall = 'xsmall',
+}
+
+const buttonSizeToIconSize: Record<ButtonSize, IconSize> = {
+  [ButtonSize.XLarge]: IconSize.XLarge,
+  [ButtonSize.Large]: IconSize.Large,
+  [ButtonSize.Medium]: IconSize.Medium,
+  [ButtonSize.Small]: IconSize.Small,
+  [ButtonSize.XSmall]: IconSize.XSmall,
+};
+
+export type IconType = React.ReactElement<IconProps>;
+
+export { ButtonColor };
+
+export enum ButtonVariant {
+  Primary = 'primary',
+  Secondary = 'secondary',
+  Tertiary = 'tertiary',
+  Float = 'tertiaryFloat',
+}
```

Comment:
We currently also have `btn-quaternary` as a class what should happen to that one?

Reply:
I'd like to keep it separate from this PR, as we have the QuarternaryButton component. Ideally, we create a followup PR where the QuarternaryButton and ClickableText components are merged with the Button component and we decide via variant which type of button we want to render.

Reply:
I mean sometimes we use `Quarternarybutton` which is a seperate component.
But I see we also use `<Button className="btn-quarternary" />`

Just something to check perhaps?

Reply:
I checked the existing usage of `btn-quaternary`. Other than in QuaternaryButton, it's used in `MultiLevelMenuDetail` component and in `EmailForm`. In both cases, removing the `btn-quaternary` class does not have any effect on the appearance of the button. I think we can ignore that for now and tackle it when doing refactoring of the Button usage in a followup PR.

Reply:
Perfect, thanks for checking, so we can pretty much just remove it there then.